### DISCUSSION TITLE
Wrap `.reset` function with `useCallback`

### DISCUSSION THREAD

**Discussion by vlad-khitev-axon** [next-safe-action/next-safe-action#179] [Ideas]
It seems that in v6 the reference to the [`reset`](https://github.com/TheEdoRan/next-safe-action/blob/5b9ca528a68eed6c0b54d76e6d0dd4e55084270a/packages/next-safe-action/src/hooks.ts#L102-L106) function was stable, so we could safely use it as an input to `useEffect` (e.g. to reset the action state when the modal window is closed). Currently, this function is recreated on each render, which looks like a breaking change. Could we roll it back?

```tsx
const reset = useCallback(() => {
  setIsIdle(true);
  setClientInput(undefined);
  setResult({});
}, []);
```

**Comment by TheEdoRan** [comment]
Fixed in [v7.1.2](https://github.com/TheEdoRan/next-safe-action/releases/tag/v7.1.2)
